File: ObserverPattern/Java/NewsAgency/src/com/laxmena/News.java
package com.laxmena;

import java.time.LocalDateTime;

public class News {
    private String title;
    private String content;
    private LocalDateTime publishedOn;
    private String publishedBy;

    public News(String title, String content, LocalDateTime date, String publisher) {
        this.title = title;
        this.content = content;
        this.publishedOn = date;
        this.publishedBy = publisher;
    }

    public String getTitle() {
        return title;
    }

    public String getContent() {
        return content;
    }

    public String getPublishedBy() {
        return publishedBy;
    }

    public LocalDateTime getPublishedOn() {
        return publishedOn;
    }

    public void printFormattedNews() {
        System.out.println("------------");
        System.out.println("Published by: " + this.publishedBy + " | Published on: " + this.publishedOn.toString());
        System.out.println("Title: " + this.title);
        System.out.println("Content: " + this.content);
        System.out.println("------------");
    }
}

File: Observer/Python/NewsAgency/News/news.py
import string


class News:
    title: string = None
    content: string = None
    published_on: string = None
    published_by: string = None

    def __init__(self, title: string, content: string, date: string, author: string):
        self.title = title
        self.content = content
        self.published_on = date
        self.published_by = author

    def get_title(self):
        return self.title

    def get_content(self):
        return self.content

    def get_published_by(self):
        return self.published_by

    def get_published_on(self):
        return self.published_on

    def print_formatted_news(self):
        print("--------------")
        print("Published by: {} | Date: {}".format(self.published_by, self.published_on))
        print("Title: {}".format(self.title))
        print("Content: {}".format(self.content))
        print("--------------")
        